topics/dev/tutorials/debugging/tutorial.md

```diff
@@ -0,0 +1,730 @@
+>    >  
+>    >  Try to identify which tests are failing based on artifacts you downloaded.
+>    >  
+>    >  > ### {% icon solution %} Solution
+>    >  >
+>    >  > The failing tests are:
+>    >  >
+>    >  > - `test_build_list_and_show_items`
+>    >  > - `test_data_column_input`
+>    >  > - `test_upload_pair_specify_extension`
+>    >  {: .solution }
+>    >  
+> 2. __Running the tests locally__
+>    >  
+>    >  We do not want to run the full suite of Selenium tests here - that would take a lot of time and we already know that most of them passed.
+>    >  
+>    >  Instead, we can build our commands to only run a subset of the tests. Here's an example: `./run_tests.sh -selenium lib/galaxy_test/selenium/test_sign_out.py`
+>    >  
+>    >  A command like that will still run all of the tests defined in that python file, so we need to find out where the tests that failed are defined. Try seaching the baseline for our three failing Selenium tests; what files are they in?
+>    >  
+>    >  > ### {% icon solution %} Solution
+>    >  >
+>    >  > The failing tests are defined in these files:
+>    >  >
+>    >  > - `lib/galaxy_test/selenium/test_collection_builders.py`
+>    >  > - `lib/galaxy_test/selenium/test_workflow_editor.py`
+>    >  > - `lib/galaxy_test/selenium/test_uploads.py`
+>    >  {: .solution }
+>    >  
+>    >  Now that we have the files, we can run the selenium tests. The command to do this should be very similar to `./run_tests.sh -selenium lib/galaxy_test/selenium/test_sign_out.py`. The only difference should be the python file where the tests are defined. 
+>    >  
+>    >  Running any of these three tests will show us that the tests are still failing on our local machine. So, you should see a failing test that matches one of the three we identified from the downloaded artifacts. Onward to fixing them!
+>    >  
+> 3. __Finding and fixing the issue__
+>    >  
+>    >  Good news - there's an easy way to figure out what's going wrong with a selenium test! In your terminal type `export GALAXY_TEST_SELENIUM_HEADLESS=false`. This will allow you to see the test running in real time. Re-run the selenium test to watch the headless browser step through the test. Looking at the test run in the headless browser, alongside the definition for the test should help you understand what might be going wrong. Try and figure out where the problem is on your own, before looking at the solution.
+>    >  
+>    >  > ### {% icon solution %} Solution
+>    >  >
+>    >  > The problem is that the tests are expecting a different HID than what is being made available to them.  Since all the failing tests have to do with Lists, you might want to check out the ListCollectionCreator.
```

Comment:
```suggestion 
The problem is that the tests are expecting a different HID (history identifier) than what is being made available to them.  Since all the failing tests have to do with Lists, you might want to check out the ListCollectionCreator.
```

Reply:
Applied. Good point!